Redesigned welcomescreen and added help button

jriyyya opened the pull request:
@chimosky @sourabhaa Please Review

jriyyya committed:
Redesigned welcomescreen and added help btn

- The UI for welcomescreen now has more visuals

- There is a help button which shows how to play the game

- An animated jumping sonic is added to welcomescreen

- The help screen shows keys to use and double jump mechanic

Dimi20cen commented:
My first attempt to review, feedback is appreciated
Tested with pygame 2.4.0 and python 3.7.3

- Arrow key for jumping is not working, however same issue exists in master branch as well, in my case it can be fixed by updating sugargame to its latest version. If I understand correctly, related with #4

jriyyya commented:
Should we then open an issue for this or we can even mention it in #4 as it is unrelated to this PR

Dimi20cen commented:
I believe that makes sense, yes. I'll open a separate PR.

chimosky commented on welcomescreen.py:
```diff
@@ -126,10 +135,36 @@ def run(self, gameDisplay):
 
-        # GAME LOOP BEGINS !!!
+        sonic1 = pygame.image.load("images/sonic.png")
+        t = sonic1.get_rect()
```
t?

jriyyya replied:
sonic1_rect will be better 😅

chimosky commented on welcomescreen.py:
```diff
@@ -126,10 +135,36 @@ def run(self, gameDisplay):
+        def get_sonic_pos():
+            if abs(self.step) < width / 8:
+                self.step += 3
+            elif abs(self.step) < width / 4:
```
Use floor division for consistency.

chimosky commented on welcomescreen.py:
```diff
@@ -126,10 +135,36 @@ def run(self, gameDisplay):
+                self.step += 5
+            if self.step < -width // 2 or self.step > width // 2:
+                self.step = -width // 2
+            x = startx + width / 2 + self.step - sonic1.get_rect().width / 2
```
Floor division here too.

chimosky commented on welcomescreen.py:
```diff
@@ -146,41 +181,68 @@ def run(self, gameDisplay):
+                self.showing_help = False
+
+            # Help Button Action
+            if help_btn.collidepoint(mos_x, mos_y) and (pygame.mouse.get_pressed())[0] == 1 and press == 0 and not self.showing_help:
```
Wanna break this to be pep8 compliant?

jriyyya replied:
Indeed, Done

chimosky commented:
Tested b533f86, looks good. Thanks!

jriyyya committed:
Merge branch 'sugarlabs:master' into patch-ui

jriyyya commented:
@chimosky Can we merge this?

chimosky commented:
I thought I'd made my comments when I reviewed this earlier but it seems I didn't.

jriyyya committed:
change variable name t to sonic1_rect

jriyyya committed:
change division to floor division

jriyyya committed:
help button action condition pep8 compliant

jriyyya commented:
I see, Done the requested changes.

chimosky commented on welcomescreen.py:
```diff
@@ -226,7 +226,9 @@ def get_sonic_pos():
             # Help Button Action
-            if help_btn.collidepoint(mos_x, mos_y) and (pygame.mouse.get_pressed())[0] == 1 and press == 0 and not self.showing_help:
+            if help_btn.collidepoint(mos_x, mos_y) and \
+               (pygame.mouse.get_pressed())[0] == 1 and \
+               press == 0 and not self.showing_help:
```
The indentation here is wrong, it should be in a way that the block below the if statement shouldn't be difficult to find.

jriyyya replied:
```
            if help_btn.collidepoint(mos_x, mos_y) and \
                                    (pygame.mouse.get_pressed())[0] == 1 and \
                                    press == 0 and not self.showing_help:
                self.showing_help = True
```

Is this a correct way?

chimosky replied:
The indentation should raise flake8 errors so no, something like this;

```python3
if help_btn.collidepoint(mos_x, mos_y) and \
        (pygame.mouse.get_pressed())[0] == 1 and \
        press == 0 and not self.showing_help:
    self.showing_help = True
```

jriyyya replied:
Hmm, This looks much better. Done

chimosky commented on welcomescreen.py:
```diff
@@ -229,7 +229,7 @@ def get_sonic_pos():
                (pygame.mouse.get_pressed())[0] == 1 and \
                press == 0 and not self.showing_help:
-                self.showing_help = True
+                    self.showing_help = True
```
This is supposed to be the block under the if condition and it's way more than 4 spaces.

jriyyya replied:
Removed this commit through force push

chimosky commented:
Would be great to include the reason why floor division isn't used in your commit message in bff13bf.

jriyyya committed:
Correct indentation of if condition of help button action

jriyyya committed:
Remove floor divison in scale_fac

- Changing it into floor divison results in making its
  value 0 as it less than 1, hence where ever the scale_fac
  is multiplied, those objects are rendered on top of the
  screen instead of its desired position

jriyyya commented:
> Would be great to include the reason why floor division isn't used in your commit message in [bff13bf](https://github.com/sugarlabs/sonic-jump-activity/commit/bff13bf76e1ac53cbe1fb375c84159aefcc31440).

Thanks, Specified it.
